### ISSUE TITLE
Unstable AR Content and Ineffective Tracking Configurations in MindAR

### ISSUE THREAD

**Issue by param-fsd** [hiukim/mind-ar-js#556]
The AR content (e.g., 3D models, videos) rendered using MindAR on markers is highly unstable. Despite tweaking tracking configurations (filterMinCF, filterBeta, etc.), the instability persists, resulting in jittery content that negatively impacts the user experience.

**Expected Behavior**
The AR content should stabilize on the marker without noticeable jitter, even during small movements of the marker or camera.

**Request for Help**
Tracking configurations are insufficient to solve this issue. Could someone assist in implementing custom stabilization functionality, such as:
**Custom Smoothing Algorithms: Smooth AR content position and rotation.**
- Device Orientation Control: Stabilize AR objects based on device orientation sensors.
- Hybrid Tracking Solutions: Combine marker-based tracking with external tracking methods for improved stability.
- Any code snippets, references, or detailed guidance would be greatly appreciated!

**Comment by dogzilla** [comment]
Personally, I’ve found all JS AR solutions to be really sensitive to lighting and target image. Have you experimented with changing the target up (if possible)?Sent from my iPhoneOn Dec 15, 2024, at 3:55 AM, PS Prashant ***@***.***> wrote:﻿
The AR content (e.g., 3D models, videos) rendered using MindAR on markers is highly unstable. Despite tweaking tracking configurations (filterMinCF, filterBeta, etc.), the instability persists, resulting in jittery content that negatively impacts the user experience.
Expected Behavior
The AR content should stabilize on the marker without noticeable jitter, even during small movements of the marker or camera.
Request for Help
Tracking configurations are insufficient to solve this issue. Could someone assist in implementing custom stabilization functionality, such as:
Custom Smoothing Algorithms: Smooth AR content position and rotation.

Device Orientation Control: Stabilize AR objects based on device orientation sensors.
Hybrid Tracking Solutions: Combine marker-based tracking with external tracking methods for improved stability.
Any code snippets, references, or detailed guidance would be greatly appreciated!

**Comment by param-fsd** [comment]
Hello [dogzilla](https://github.com/dogzilla), 
In 8th Wall and MyWebAR, the tracking was actually good. 
I observed that they apply smoothing to the camera movement and content. If it's possible to add a custom function for smoothing both the camera and content, it would be really helpful, and I’m looking forward to a solution for the same.

I also tested this using high-textured patterns, but the same issue persists.

**Comment by param-fsd** [comment]
https://github.com/user-attachments/assets/f8c9f4f0-e1fc-4ed2-acb3-2d6164d1ba1e

**Comment by dogzilla** [comment]
Yes, 8thWall and other commercial solutions do a decent job - I’m sure they have tuned the models that do the tracking. I don’t believe the MindAR tracking models have been significantly updated since initial release 3 years ago, but maybe someone else can speak directly to that.Have you seen the smoothing parameters for MindAR? They help quite a bit, although I haven’t been able to get the tracking as smooth as what I see in 8thWall yet. Here’s the docs on that: https://hiukim.github.io/mind-ar-js-doc/quick-start/tracking-config/ 

That said, 8thWall has a minimum license fee of $750/month per project, MyWebAR starts at $799/mo total for something comparable to the flexibility of MindAR - those prices are well out of the realistic for many people (me included), given AR is a relatively new and unproven tech. Given the option between those prices and jitter, all of my clients so far have chosen to live with the jitter.

**Comment by YulesRules** [comment]
First off, Mind-ar is a great achievement. However, as of writing, I haven't seen any open source libraries which handle image tracking in as robust a fashion as traditional fiducial markers with borders. This makes the technology relatively unusable outside of subscribing to third party services who integrate multiple novel approaches for stability and reliability. I'm sure this will change as the web XR standards get adopted over time and we see approaches also integrating visual SLAM and IMU data (like 8th Wall).

Until then, you have to choose your battles very carefully to ensure the markers and environments are 100% ideal.
